# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/12/30 19:00:56+01:00 bzolnier@trik.(none) 
#   [ide] it8172: incorrect return from it8172_init_one()
#   
#   From: Alan Cox <alan@lxorguk.ukuu.org.uk>
#   
#   Several IDE drivers return positive values as errors in the PCI setup
#   code. Unfortunately the PCI layer considers positive values as success
#   so the driver skips the device but still claims it and things then go
#   downhill.
#   
#   This fixes the IT8172 driver.
#   
#   From: Francois Romieu <romieu@fr.zoreil.com>
#   
#   Use -ENODEV instead of -EAGAIN.
#   
#   Signed-off-by: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
# 
# drivers/ide/pci/it8172.c
#   2004/12/30 19:00:46+01:00 bzolnier@trik.(none) +1 -1
#   [ide] it8172: incorrect return from it8172_init_one()
# 
diff -Nru a/drivers/ide/pci/it8172.c b/drivers/ide/pci/it8172.c
--- a/drivers/ide/pci/it8172.c	2005-01-22 23:08:16 -08:00
+++ b/drivers/ide/pci/it8172.c	2005-01-22 23:08:16 -08:00
@@ -270,7 +270,7 @@
 {
         if ((!(PCI_FUNC(dev->devfn) & 1) ||
             (!((dev->class >> 8) == PCI_CLASS_STORAGE_IDE))))
-                return 1; /* IT8172 is more than only a IDE controller */
+		return -ENODEV; /* IT8172 is more than an IDE controller */
 	ide_setup_pci_device(dev, &it8172_chipsets[id->driver_data]);
 	return 0;
 }
